### ISSUE TITLE
Not compiling at Ubuntu 20.04

### ISSUE THREAD

**Issue by xepost** [nitroshare/nitroshare-desktop#248]
Hi 

I  wanted to install but there are some minor issues.  

First, there are some missing packages, so far I have installed the following packages.

```
sudo apt install  build-essential cmake
sudo apt install qttools5-dev-tools  qttools5-dev
sudo apt install libqt5svg5
sudo apt install libqt5svg5-dev 
sudo apt install appindicator3-sharp 
sudo apt install  libappindicator0.1-cil-dev 
sudo apt install libnotify4 
sudo apt install libnotify-dev 
sudo apt install indicator-multiload
sudo apt install libgtk2.0-dev
```

I still get warnings for  2 missing packages, any suggestions would be appreciated.

```
user@user:~/Desktop/nitroshare-0.3.4/build$ cmake  ..
CMake Warning at CMakeLists.txt:44 (message):
  QHttpEngine not found - local API disabled


CMake Warning at CMakeLists.txt:52 (message):
  QMdnsEngine not found - local API disabled


-- Checking for modules 'gtk+-2.0;appindicator-0.1;libnotify'
--   No package 'gtk+-2.0' found
--   No package 'appindicator-0.1' found
-- Configuring done
-- Generating done
-- Build files have been written to: /home/user/Desktop/nitroshare-0.3.4/build
```

The second issue is related with the Qt5 version. 
```
[  2%] Automatic MOC for target nitroshare
[  2%] Built target nitroshare_autogen
[  4%] Building CXX object src/CMakeFiles/nitroshare.dir/application/splashdialog.cpp.o
/home/user/Desktop/nitroshare-0.3.4/src/application/splashdialog.cpp: In constructor ‘SplashDialog::SplashDialog()’:
/home/user/Desktop/nitroshare-0.3.4/src/application/splashdialog.cpp:40:25: error: incomplete type ‘QStyle’ used in nested name specifier
   40 |     setGeometry(QStyle::alignedRect(Qt::LeftToRight, Qt::AlignCenter, size(), QApplication::desktop()->availableGeometry()));
      |                         ^~~~~~~~~~~
make[2]: *** [src/CMakeFiles/nitroshare.dir/build.make:183: src/CMakeFiles/nitroshare.dir/application/splashdialog.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:97: src/CMakeFiles/nitroshare.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

The solution can be seen here

https://github.com/Nitrokey/nitrokey-app/issues/361#issuecomment-394352608
Basically  just add the
```
#include <Style>
 ```
inside the splashdialog.cpp by your  favorite text editor
gedit  /home/user/Desktop/nitroshare-0.3.4/src/application/splashdialog.cpp

**Comment by mpiffault** [comment]
For the record, it's `#include <QStyle>` (instead of Style)

**Comment by answerquest** [comment]
Any way forward found for `QHttpEngine not found - local API disabled` ?

**Comment by geroldr** [comment]
> Any way forward found for `QHttpEngine not found - local API disabled` ?

On Debian 11 I had to install libqhttpengine-dev. This will pull in libqhttpengine0.
